### ISSUE TITLE
SO_REUSEPORT is not handled correctly

### ISSUE THREAD

**Issue by dmlloyd** [quarkusio/quarkus#7493]
**Describe the bug**
The Quarkus Vert.x configuration is mishandling `SO_REUSEPORT`.  It is being treated as if it were just a switch you can throw on a socket and get some effect, but this is not the case.  This option requires correct handling, and furthermore we should always be enabling it because it can substantially impact performance.

**Expected behavior**
When `SO_REUSEPORT` is active (which should be always), Vert.x should be creating and managing one `*ServerSocketChannel` instance *per event loop* (that is, creating as many sockets as there are event loops) with multiple event loops active (one per core).  When `SO_REUSEPORT` is disabled, then the existing behavior should remain.

**Actual behavior**
`SO_REUSEPORT` appears to be enabled on the server socket uselessly, with only one server socket channel being used.

**Additional context**
`SO_REUSEPORT` is only supported on Java 9+ or on the Netty native transport.  #5691 may be a prerequisite in this regard unless we add some kind of detection.

This probably merits an upstream Vert.x issue but I wanted to check with Julien first.

/cc @vietj.

**Comment by vietj** [comment]
server socket sharing is a feature of Vert.x component model (verticles) for enabling load balancing based on the number of verticles which gives control over the number of cores you want to use for a server.

`SO_REUSEPORT` was added later as a side effect of handling Netty native transports mainly.

We did not consider using it to achieve the same on os-able and JDK-able systems. I believe for Vert.x 4 we should consider using `SO_REUSEPORT` when available to achieve the same feature and let the OS do the job otherwise use the current server socket sharing.
